store/tikv/region_cache.go

```diff
@@ -1768,6 +1770,13 @@ func (s *Store) initResolve(bo *Backoffer, c *RegionCache) (addr string, err err
 	}
 }
 
+// A quick and dirty solution to found whether an err is caused by StoreNotFound.
+// todo: A better solution, maybe some err-code based error handling?
+func isStoreNotFoundError(err error) bool {
+	return err.Error() == "[pd] store field in rpc response not set" ||
+		(strings.Contains(err.Error(), "invalid store ID") && strings.Contains(err.Error(), "not found"))
```

Comment:
```suggestion
	return strings.Contains(err.Error(), "invalid store ID") && strings.Contains(err.Error(), "not found")
```

Reply:
Confirmed with PD members. `"[pd] store field in rpc response not set"` may happen when the leader of PD is changed.